src/core/mac/mac.cpp

```diff
@@ -856,10 +856,116 @@ void Mac::HandleBeginTransmit(void)
 
     if (error != kThreadError_None)
     {
+#if OPENTHREAD_ENABLE_PASSING_UP_ACK
+        TransmitDoneTask(mTxFrame, NULL, kThreadError_Abort);
+#else
         TransmitDoneTask(mTxFrame, false, kThreadError_Abort);
+#endif
+    }
+}
+
+#if OPENTHREAD_ENABLE_PASSING_UP_ACK
```

Comment:
Yes, the config is a temporary solution, since not all platforms could pass up ACK frame for now.